lib/irq.c

```diff
@@ -236,28 +256,46 @@ static int
 irqs_set_data_eventfd(vfu_ctx_t *vfu_ctx, struct vfio_irq_set *irq_set,
                       int *data)
 {
-    int efd;
+    int *efd;
     uint32_t i;
     size_t j;
 
     assert(data != NULL);
     for (i = irq_set->start, j = 0; i < (irq_set->start + irq_set->count);
          i++, j++) {
-        efd = vfu_ctx->irqs->efds[i];
-        if (efd >= 0) {
-            if (close(efd) == -1) {
-                vfu_log(vfu_ctx, LOG_DEBUG, "failed to close IRQ fd %d: %m", efd);
-            }
-
-            vfu_ctx->irqs->efds[i] = -1;
+        switch (irq_set->index) {
+        case VFIO_PCI_ERR_IRQ_INDEX:
+            efd = &vfu_ctx->irqs->err_efd;
+            break;
+        case VFIO_PCI_REQ_IRQ_INDEX:
+            efd = &vfu_ctx->irqs->req_efd;
+            break;
+        default:
+            efd = &vfu_ctx->irqs->efds[i];
+            break;
+        }
+        if (*efd >= 0) {
+            if (close(*efd) == -1)
```

Comment:
missing curly brackets

Reply:
Yes, this is libvfio-user's coding style.

Reply:
Ok.